SU2_CFD/src/driver_structure.cpp

```diff
@@ -3642,6 +3886,244 @@ void CFluidDriver::SetInitialMesh() {
   //}
 }
 
+CTurbomachineryDriver::CTurbomachineryDriver(char* confFile,
+    unsigned short val_nZone,
+    unsigned short val_nDim, SU2_Comm MPICommunicator) : CFluidDriver(confFile,
+        val_nZone,
+        val_nDim,
+        MPICommunicator) { }
+
+CTurbomachineryDriver::~CTurbomachineryDriver(void) { }
+
+void CTurbomachineryDriver::Run() {
+
+
+//  unsigned long ExtIter = config_container[ZONE_0]->GetExtIter();
+
+  int rank = MASTER_NODE;
+
+#ifdef HAVE_MPI
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+#endif
+
+
+  /*--- Run a single iteration of a multi-zone problem by looping over all
+   zones and executing the iterations. Note that data transers between zones
+   and other intermediate procedures may be required. ---*/
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    iteration_container[iZone]->Preprocess(output, integration_container, geometry_container,
+                                           solver_container, numerics_container, config_container,
+                                           surface_movement, grid_movement, FFDBox, iZone);
+  }
+
+  /* --- Update the mixing-plane interface ---*/
+  for (iZone = 0; iZone < nZone; iZone++) {
+    if(mixingplane)SetMixingPlane(iZone);
+  }
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    iteration_container[iZone]->Iterate(output, integration_container, geometry_container,
+                                        solver_container, numerics_container, config_container,
+                                        surface_movement, grid_movement, FFDBox, iZone);
+  }
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    iteration_container[iZone]->Postprocess(config_container, geometry_container,
+                                            solver_container, iZone);
+  }
+
+  if (rank == MASTER_NODE){
+    SetTurboPerformance(ZONE_0);
+  }
+
+
+}
+
+void CTurbomachineryDriver::SetMixingPlane(unsigned short donorZone){
```

Comment:
I see, so maybe we should change the name of the class to something else.
Indeed "interpolation" (in a strictly mathematical sense) doesn't actually occur even for the nearest neighbour approach (zones are loosely coupled and the interface is basically treated as a non-homogeneous BC).
Having all the possible way of communicating between zones in the same src  would make the code much more clear, and possibly we could avoid creating the new driver class just by modifying the fluid driver a little bit.
By the way, we may do this in a later pull request (so nevermind) I'm just trying to figure out if that would make sense.